razorpay-payments.php

```diff
@@ -133,53 +134,116 @@ function receipt_page($order)
             echo $this->generate_razorpay_form($order);
         }
 
+        protected function getSessionKey($orderId)
+        {
+            return "razorpay_order_id.$orderId";
+        }
+
         /**
          * Generate razorpay button link
          **/
-        public function generate_razorpay_form($order_id)
+        public function generate_razorpay_form($orderId)
         {
             global $woocommerce;
-            $order = new WC_Order($order_id);
+            $order = new WC_Order($orderId);
+
             $redirect_url = get_site_url() . '/?wc-api=' . get_class($this);
-            $productinfo = "Order $order_id";
+            $productinfo = "Order $orderId";
+            $api = new Api($this->key_id, $this->key_secret);
+
+            $sessionKey = $this->getSessionKey($orderId);
+
+            $verify= true;
+
             try
             {
-                $api = new Api($this->key_id, $this->key_secret);
-                // Calls the helper function to create order data
-                $data = $this->get_order_creation_data($order_id);
-
-                $razorpay_order = $api->order->create($data);
-
-                $woocommerce->session->set('razorpay_order_id', $razorpay_order['id']);
-                $razorpay_args = array(
-                  'key'             => $this->key_id,
-                  'name'            => get_bloginfo('name'),
-                  'amount'          => $order->order_total * 100,
-                  'currency'        => get_woocommerce_currency(),
-                  'description'     => $productinfo,
-                  'prefill'         => array(
-                    'name'              => $order->billing_first_name." ".$order->billing_last_name,
-                    'email'             => $order->billing_email,
-                    'contact'           => $order->billing_phone
-                  ),
-                  'notes'           => array(
-                    'woocommerce_order_id' => $order_id
-                  ),
-                  'order_id'        => $razorpay_order['id']
-                );
-                $json = json_encode($razorpay_args);
-                $html = $this->generate_order_form($redirect_url, $json);
-                return $html;
+                if ($woocommerce->session->get($sessionKey))
+                {
+                    $razorpayOrderId = $woocommerce->session->get($sessionKey);
+                    $verify = $this->verifyOrderAmount($razorpayOrderId, $orderId);
+                }
+                else if ($verify === false)
```

Comment:
@mayankamencherla The `else if` block will never be called, because verify is set to false _inside the if_.